### ISSUE TITLE
Problem upgrading JHipster app (4.6 -> 5.7)

### ISSUE THREAD

**Issue by JordySwinnenPXL** [jhipster/generator-jhipster#8952]
##### **Overview of the issue**

Generated an application with JHipster in a older version.
Trying to upgrade to latest JHipster version

##### **Motivation for or Use Case**

Can not upgrade JHipster application to latest version

##### **Reproduce the error**
`npm install generator-jhipster@4.6.0`
Generate JHipster application.
Import simple JDL.
Manualy add simple code.
`npm install generator-jhipster@latest`
Do `jhipster upgrade --verbose`
Result:
```
events.js:167
      throw er; // Unhandled 'error' event
      ^

Error: ERROR! Exiting process
    at Environment.error (%AppData%\Roaming\npm\node_modules\generator-jhipster\node_modules\yeoman-environment\lib\environment.js:157:40)
    at module.exports.error (%AppData%\Roaming\npm\node_modules\generator-jhipster\generators\generator-base.js:2444:18)
    at gitExec (%AppData%\Roaming\npm\node_modules\generator-jhipster\generators\upgrade\index.js:235:30)
    at %AppData%\Roaming\npm\node_modules\generator-jhipster\node_modules\shelljs\src\exec.js:118:9
    at ChildProcess.exithandler (child_process.js:280:7)
    at ChildProcess.emit (events.js:182:13)
    at maybeClose (internal/child_process.js:962:16)
    at Socket.stream.socket.on (internal/child_process.js:381:11)
    at Socket.emit (events.js:182:13)
    at Pipe._handle.close (net.js:606:12)
Emitted 'error' event at:
    at Environment.error (%AppData%\Roaming\npm\node_modules\generator-jhipster\node_modules\yeoman-environment\lib\environment.js:159:15)
    at module.exports.error (%AppData%\Roaming\npm\node_modules\generator-jhipster\generators\generator-base.js:2444:18)
    [... lines matching original stack trace ...]
    at Pipe._handle.close (net.js:606:12)
```

Tried reinstalling with this command:
`rm -rf node_modules && npm uninstall generator-jhipster -g && npm cache clean --force && npm install generator-jhipster@latest -g && npm install && jhipster upgrade --verbose`

Not exactly sure what is going wrong here since the error log is unclear to me, hence this post.

##### **Related issues**

None found

##### **JHipster Version(s)**

JHipster 4.6 -> 5.7

##### **JHipster configuration**

[Github Gist of jhipster info](https://gist.github.com/JordySwinnenPXL/95c5975c5621c82e9806143af8760dc5)

##### **Browsers and Operating System**

Windows 10

-   [x] Checking this box is mandatory (this is just to show you read everything)

EDIT: 
##### **Similair issues**
https://github.com/jhipster/generator-jhipster/issues/6161

**Comment by jdubois** [comment]
I'm testing this ASAP

**Comment by jdubois** [comment]
Anyway there are so many changes between those 2 major versions that you should expect quite a bit of manual work. I remember it took 2 days to my trainees to migrate JHipster Online to JHipster 5.

**Comment by JordySwinnenPXL** [comment]
> I'm sorry I can't reproduce, but this looks like an installation error with NPM - I've seen this quite often in dev mode, using "npm link".
> 
> This depends on how you installed JHipster, but usually I need to do a "npm install" inside the "generator-jhipster" module (this is like transitive dependencies are broken). This was happening very often with Yarn, which is one of the big reasons we stopped using it, and as you are on JHipster 4 and that Yarn was the default, there are high chances this is your issue. Maybe you could switch to NPM first?? This is a "flag" at generation time, and you can also find this in your .yo-rc.json configuration (just change it and regenerate to migrate).

Hey, thank you for the response.

Even though i don't use yarn (always used npm), i checked my .yo-rc.json file to be sure. I assume you were talking about this line `"clientPackageManager": "npm",` but it was already set to npm in my project.

I have also tried running `npm install` inside %AppData%\Roaming\npm\node_modules\generator-jhipster but without succes unfortunately.

After that i tried upgrading to a diffrent JHipster version.
`npm install generator-jhipster@4.7.0`
`jhipster upgrade --verbose --target-version=4.7.0`
But without succes.

tl:dr => So far no solution. But i will post here when and if find one.

**Comment by aymenbraiek** [comment]
fdefef

**Comment by gmarziou** [comment]
Error shows `%AppData%`, I'm a bit surprised to see the environment variable not expanded so I'm wondering which Windows shell are you using to run `jhipster` command.

**Comment by smlshn** [comment]
I had the same issue tonight, thanks you @gmarziou, solved using your advise. Let me define what I have done so far. 
1. I had 5.7.0 installed on my mac
2. I upgraded it to 5.7.2 and suddenly got that error
3. `rm -rf /usr/local/lib/node_modules/generator-jhipster` 
4. npm install -g generator-jhipster@5.7.0
then error gone.

**Not**: I tried to upgrade jhipster version, so I will try to do the same for the latest version see if it will work.

**Comment by pascalgrimaud** [comment]
Looks like it is solved. So I'm closing this.
